packages/shopping/src/controllers/user-order.controller.ts

```diff
@@ -35,10 +37,20 @@ export class UserOrderController {
       },
     },
   })
+  @authenticate('jwt')
   async createOrder(
     @param.path.string('userId') userId: string,
+    @inject('authentication.currentUser') currentUser: UserProfile,
     @requestBody() order: Order,
   ): Promise<Order> {
+    if (currentUser.id !== order.userId) {
```

Comment:
The rewrite here is different from the expected behaviour:

The expected requirement mentioned [here](https://github.com/strongloop/loopback4-example-shopping/issues/43#issuecomment-468311127) is *don't take in `userId` from request*:

```ts
@authenticate('jwt')
async createOrder(
  @inject('authentication.currentUser') currentUser: UserProfile,
  @requestBody() order: Order,
) {
  if (!currentUser.id) {
    throw new HttpErrors.BadRequest(`No user logged in.`);
  }
  // The rest of code that creates orders for the logged in user.
}
```

But your logic which compares the logged in user id and the id from query also makes sense to me.

If aligning with the original requirements costs too much effort I am ok with the approach here.

Any thought @strongloop/loopback-next ?

Reply:
+1. Agree. No user id in request.

Reply:
Yes please, we should really remove `userId` argument from this controller method.

In the implementation proposed by this pull request, there are too many places where user id can be specified:
 - `userId` as a path parameter
 - `order.userId` from request body
 - `currentUser.id` from the authentication layer